components/raftstore/src/store/peer.rs

```diff
@@ -2617,15 +2678,19 @@ impl Peer {
         None
     }
 
-    fn region_replication_status(&mut self) -> Option<RegionReplicationStatus> {
+    fn region_replication_status<T, C>(
+        &mut self,
+        ctx: &PollContext<T, C>,
+    ) -> Option<RegionReplicationStatus> {
         if self.replication_mode_version == 0 {
             return None;
         }
         let mut status = RegionReplicationStatus::default();
         status.state_id = self.replication_mode_version;
         let state = if !self.replication_sync {
             if self.dr_auto_sync_state != DrAutoSyncState::Async {
-                let res = self.raft_group.raft.check_group_commit_consistent();
+                let res = self.check_group_commit_consistent(ctx.cfg.group_consistent_log_gap);
+                self.check_wait_sync_deadline(res.unwrap_or(false), ctx.get_current_time());
```

Comment:
This works at minute cycles, may not be frequent enough.

Reply:
I had added the check to raft tick, PTAL.